stdlib/public/core/Flatten.swift.gyb

```diff
@@ -123,6 +123,17 @@ def collectionForTraversal(traversal):
     assert False, "unknown traversal"
 }%
 
+%{
+def indexConfromsToForTraversal(traversal):
+    if traversal == 'Forward':
+        return 'Comparable'
+    if traversal == 'Bidirectional':
+        return 'Comparable'
+    if traversal == 'RandomAccess':
+        return 'Strideable'
+    assert False, "unknown traversal"
+}%
+
 % for traversal in ('Forward', 'Bidirectional'):
 %   t = '' if traversal == 'Forward' else traversal
 %   Collection = 'Flatten%sCollection' % t
```

Comment:
We don't use `ForwardCollection` as the protocol name, so it would be good to change `FlattenForwardCollection` to match and drop `Forward`.